### PR TITLE
WebDiscover: Finish auto deploy screen (iam configure script)

### PR DESCRIPTION
**Pull request by kimlisa**
do not merge until the following has merged:
- https://github.com/gravitational/teleport/pull/28436
- https://github.com/gravitational/teleport/pull/28537
- https://github.com/gravitational/teleport/pull/28807
- [x] tested flow without problems

This pr:
- enables the auto deploy flow
- defines the script endpoint
- emit events that differentiates between `manual` and `auto` deploy
- previously, we were passing around `integration.name` but now we pass around the full object because we also need to access the `roleArn` for the cfg script

### CODE REVIEW

**Comment by kimlisa** [comment]
friendly ping @ryanclark @ibeckermayer

File: web/packages/teleport/src/Discover/useDiscover.tsx [hunk 1]
```diff
@@ -226,9 +228,9 @@ export function DiscoverProvider({
   // The location.state.discover should contain all the state that allows
   // the user to resume from where they left of.
   function resumeDiscoverFlow() {
-    const { discover, integrationName } = location.state;
+    const { discover, integration } = location.state;
 
-    updateAgentMeta({ integrationName } as DbMeta);
+    updateAgentMeta({ integration } as DbMeta);
```
Comment (ibeckermayer) [inline web/packages/teleport/src/Discover/useDiscover.tsx:233]: The type system isn't complaining but this cast doesn't seem like it should work without making `db` and `selectedAwsRdsDb` optional.

```ts
export type DbMeta = BaseMeta & {
  // TODO(lisa): when we can enroll multiple RDS's, turn this into an array?
  // The enroll event expects num count of enrolled RDS's, update accordingly.
  db?: Database;
  integration?: Integration;
  selectedAwsRdsDb?: AwsRdsDatabase;
  // serviceDeployedMethod flag will be undefined if user skipped
  // deploying service (service already existed).
  serviceDeployedMethod?: ServiceDeployMethod;
};
```

File: web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx [hunk 2]
```diff
@@ -247,49 +257,84 @@ const CreateAccessRole = ({
...
-        placeholder="teleport"
-        width="400px"
+        placeholder="TeleportDatabaseAccess"
+        width="440px"
         mr="3"
         onChange={e => setTaskRoleArn(e.target.value)}
-        toolTipContent="Lorem ipsume dolores"
+        toolTipContent={`Amazon Resource Names (ARNs) uniquely identifies AWS \
+        resources, in this case you will be naming an IAM role that this \
+        deployed service will be using`}
```
Comment (ibeckermayer) [inline web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx:310]: Not sure if this is exactly correct

```suggestion
        toolTipContent={`Amazon Resource Names (ARNs) uniquely identify AWS \
        resources. In this case you will be giving an ARN to the IAM role that this \
        deployed service will be using`}
```

If not then

```suggestion
        toolTipContent={`Amazon Resource Names (ARNs) uniquely identify AWS \
        resources. In this case you will naming an IAM role that this \
        deployed service will be using`}
```

File: web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx [hunk 3]
```diff
@@ -227,10 +237,10 @@ const Heading = ({
...
         Teleport needs a database service to be able to connect to your
         database. Teleport can configure the permissions required to spin up an
-        ECS Fargate container (0.xxx vCPU, 1GB memory) in your Amazon account
-        with the ability to access databases in this region (
-        <Mark>{region}</Mark>). You will only need to do this once for all
-        databases per geographical region. <br />
+        ECS Fargate container (2vCPU, 4GB memory) in your Amazon account with
+        the ability to access databases in this region (<Mark>{region}</Mark>).
+        You will only need to do this once for all databases per geographical
+        region. <br />
```
Comment (ibeckermayer) [inline web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx:243]:
```suggestion
        You will only need to do this once per geographical region.<br />
```

File: web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.test.tsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    <MemoryRouter
+      initialEntries={[
+        { pathname: cfg.routes.discover, state: { entity: 'database' } },
+      ]}
+    >
+      <ContextProvider ctx={ctx}>
+        <FeaturesContextProvider value={[]}>
+          <DiscoverProvider mockCtx={discoverCtx}>
+            <PingTeleportProvider
+              interval={TEST_PING_INTERVAL}
```
Comment (ibeckermayer) [inline web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.test.tsx:202]: Will this actually try to ping something?

**Review by ibeckermayer — APPROVED** [review]

File: web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.test.tsx (hunk 4 above)
Comment (kimlisa) [reply web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.test.tsx:212]: no, the test is too short (default was like 3 seconds), though i added a success state testing that will at last hit it once

**Review by ryanclark — APPROVED** [review]

File: web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx [hunk 5]
```diff
@@ -247,49 +257,84 @@ const CreateAccessRole = ({
...
+            >
+              Amazon CloudShell
+            </Link>{' '}
+            and copy/paste the following command:
+          </Text>
+          <Box mb={2}>
+            <TextSelectCopyMulti
+              lines={[
+                {
+                  text: `sudo bash -c "$(curl '${scriptUrl}')"`,
```
Comment (marcoandredinis) [inline web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx:331]:
```suggestion
                  text: `bash -c "$(curl '${scriptUrl}')"`,
```

**Pull request merged**